src/CMakeLists.txt

```diff
@@ -103,15 +103,15 @@ endif()
 
 ################################################################################
 # libhpx
-add_library(hpx ${hpx_library_link_mode_core}
+add_library(hpx_some_name ${hpx_library_link_mode_core}
```

Comment:
`hpx_intermediate` ? `hpx_inter` ?

Reply:
I may just call it `hpx_core` for now. The name will be hidden from users so we can change it easily in the future.